src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -0,0 +1,50 @@
+package codeu.controller;
+
+    import codeu.model.store.basic.UserStore;
+    import org.junit.Before;
+    import org.junit.Test;
+    import org.mockito.Mockito;
+
+    import javax.servlet.RequestDispatcher;
+    import javax.servlet.ServletException;
+    import javax.servlet.http.HttpServletRequest;
+    import javax.servlet.http.HttpServletResponse;
+    import javax.servlet.http.HttpSession;
+    import java.io.IOException;
+
+public class ProfileServletTest {
+    private ProfileServlet profileServlet;
+    private HttpServletRequest mockRequest;
+    private HttpServletResponse mockResponse;
+    private HttpSession mockSession;
+    private UserStore mockUserStore;
+    private RequestDispatcher mockRequestDispatcher;
+
+    @Before
+    public void setup() throws IOException {
+
+        profileServlet = new ProfileServlet();
+
+        mockSession = Mockito.mock(HttpSession.class);
+        mockRequest = Mockito.mock(HttpServletRequest.class);
+
+        Mockito.when(mockRequest.getParameter("username")).thenReturn("test name");
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+        Mockito.when(mockRequest.getRequestURI()).thenReturn("/users/test username");
+        mockRequest = Mockito.mock(HttpServletRequest.class);
```

Comment:
I think this line is why your tests arent working. You are resetting mockRequest and then sending that to "doGet" in line 46. A good rule of thumb to avoid this is to make sure to do your "when/thenReturn" calls after you set up all your variables.

Reply:
I still do not get it. Can you elaborate it? I have pushed some changes, without updating unit testing.

Reply:
You are setting mockRequest twice once is like 29 and another time in like 34. The mockRequest in like 29 is what you are doing the "when/return" stuff for in line 31-34. Then you overwrite mockRequest in link 39 and send it to doGet in line 46. If your remove line 34 you should be fine.

But I was saying moving forward you should do all your instantiation before you do your "when/return" stuff.

Reply:
I removed the line 34. I will add the doPost method and push the changes.